### DISCUSSION TITLE
Installing Maven and Spring Boot CLI

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/phase-3#14] [General]
# How to install Maven and Spring Boot CLI on Windows, macOS, and Ubuntu (or other Debian OS)

## Windows
- Run Windows Terminal/Powershell (if not available, download the latest versions of **Powershell** and/or **Windows Terminal** from the Microsoft Store)
- Open a **modern** Windows Terminal or Powershell (NOT cmd), and execute the following commands to install `scoop:`
    - `Set-ExecutionPolicy RemoteSigned -scope CurrentUser` (This might not work if your execution policy is already set or if your PowerShell version is older. Latest PowerShell is recommended)
    - `Invoke-Expression (New-Object System.Net.WebClient).DownloadString('https://get.scoop.sh')`
    - `scoop install curl`
    - `Remove-item alias:curl`
    - `java -version` (Make sure Java 11+ is installed and visible)
    - `mvn -v`
    - If maven is not installed: `scoop install maven`
    - `scoop bucket add extras`
    - `scoop install springboot`

- Alternatively, if `scoop` doesn't work for you, use `chocolatey` to install `Maven` and `Spring Boot CLI`. Open a terminal or PowerShell as **administrator** and execute the following:
    - Run Windows Terminal/Powershell as administrator and execute the next set of commands
    - `Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('[https://community.chocolatey.org/install.ps1'))](https://us06st1.zoom.us/web_client/0urcsz4/html/externalLinkPage.html?ref=https://community.chocolatey.org/install.ps1%27)))`
    - `choco install maven`
    - `choco install spring-boot-cli`

## MacOS
Open a terminal and execute the following:
- `brew update`
- `brew install maven`
- `brew tap spring-io/tap` OR `brew tap pivotal/tap`
- `brew install springboot`

## Ubuntu
- `sudo apt update`
- `sudo apt upgrade`
- `sudo apt dist-upgrade`
- `sudo apt install maven`
- `mvn -version` (to check if maven was installed)
- `java -version` (to check Java 11+ is installed and visible)
- `sudo apt install unzip zip`
- `curl -s https://get.sdkman.io | bash`
- `source "/home/username/.sdkman/bin/sdkman-init.sh"=`
- `sdk flush candidates` and restart the terminal
- `sdk install springboot`
- `spring --version` (to test if Spring Boot CLI installation was successful)

**Comment by RuchikaUppal** [comment]
Getting this error while installing...
![image](https://user-images.githubusercontent.com/53488824/164615418-042cd533-0e4e-4eb2-9b82-b6627abfd976.png)

and after java - version it gives me this message

![image](https://user-images.githubusercontent.com/53488824/164615633-686297ef-7edb-4183-9d65-07a489373e5a.png)

please help

**Reply by RuchikaUppal** [reply]
The other commands are working 
![image](https://user-images.githubusercontent.com/53488824/164616067-6dbd06e0-6e01-4804-a7cf-0f42ff4d173e.png)
![image](https://user-images.githubusercontent.com/53488824/164616111-52cd6fc3-6069-4ba8-a592-697e3af3708e.png)


I hope this is ok...

**Reply by RuchikaUppal** [reply]
![image](https://user-images.githubusercontent.com/53488824/164617184-8e0a3a2a-dcce-4e25-b322-5bc95486395b.png).

How to go fwd as I followed the exact steps

**Reply by akash-coded** [reply]
@RuchikaUppal  Maybe some internal config error. I think it should work fine though. Let me know if it doesn't.

**Reply by RuchikaUppal** [reply]
Its resolved. Somehow my java got uninstalled....had to install again and set the envt and path variables. Thanks though for the reply!
